scripts/get_orphaned_functions.py

```diff
@@ -0,0 +1,243 @@
+import os
+import re
+import csv
+
+# Define Working Directory, i.e, '/pecan
+main_dir = os.getcwd()
+
+# Define a list of color codes for printing. THis proves to be useful for distinction during debugging
+color_codes = [
+    "\033[30m",  # BLACK-0
+    "\033[31m",  # RED-1
+    "\033[32m",  # GREEN-2
+    "\033[33m",  # YELLOW-3
+    "\033[34m",  # BLUE-4
+    "\033[35m",  # MAGENTA-5
+    "\033[36m",  # CYAN-6
+    "\033[37m",  # WHITE-7
+    "\033[90m",  # BRIGHT_BLACK-8
+    "\033[91m",  # BRIGHT_RED-9
+    "\033[92m",  # BRIGHT_GREEN-10
+    "\033[93m",  # BRIGHT_YELLOW-11
+    "\033[94m",  # BRIGHT_BLUE-12
+    "\033[95m",  # BRIGHT_MAGENTA-13
+    "\033[96m",  # BRIGHT_CYAN-14
+    "\033[97m",  # BRIGHT_WHITE-15
+]
+
+# Reset
+ENDC = "\033[0m"
+
+# Modules to ignore while checking for orphaned functions
+ignored_modules = ["data.hydrology", "DART"]
+
+# Function to get the list of modules
+def get_module_list():
+    module_dir = os.path.join(main_dir, "modules")
+    modules = os.listdir(module_dir)
+    return [module for module in modules if module not in ignored_modules and not module.startswith('.')]
+
+# List of modules to check for orphaned functions
+module_list = get_module_list()
+
+# Initialize two lists to store the orphaned functions
+# List_1: Purely orphaned functions
+# List_2: Functions that are utilized externally
+list_1 = []
+list_2 = []
```

Comment:
I'd suggest picking more informative variable names here. If these are named something like `funs_unused` and `funs_used_externally`, all the code below will be easier to read / edit without needing to remember whether the unused ones are list 1 or 2, and therefore less susceptible to bugs too.

Reply:
Thanks for the tip. Resolved now. Renamed them as below :
```python
purely_orphaned_functions = []
externally_utilized_functions = []
```

Would these be apt for current usage or sgould we change it to any other name?

Reply:
Those are good. Thanks!